File: app/models/torneo.py
from __future__ import annotations

from datetime import date, datetime, timezone

from sqlalchemy import Boolean, Date, DateTime, Integer, String, Text
from sqlalchemy.orm import DeclarativeBase, Mapped, mapped_column


class Base(DeclarativeBase):
    pass


class Torneo(Base):
    __tablename__ = "qnla_torneos"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    nombre: Mapped[str] = mapped_column(String(100), nullable=False)
    descripcion: Mapped[str | None] = mapped_column(Text, nullable=True)
    anio: Mapped[int] = mapped_column(Integer, nullable=False)
    fecha_inicio: Mapped[date] = mapped_column(Date, nullable=False)
    fecha_fin: Mapped[date] = mapped_column(Date, nullable=False)
    activo: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)
    cierre_inscripcion: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        default=lambda: datetime.now(timezone.utc),
        nullable=False,
    )

File: QLNA.Web/Models/Torneo.cs
using System.Text.Json.Serialization;

namespace QLNA.Web.Models;

public class Torneo
{
    [JsonPropertyName("id")]
    public int id { get; set; }
    [JsonPropertyName("nombre")]
    public string? nombre { get; set; }
    public string? descripcion { get; set; }
    [JsonPropertyName("activo")]
    public bool activo { get; set; }
}
